### ISSUE TITLE
Is it necessary to remove node globals (e.g. setImmediate) from an isolated context?

### ISSUE THREAD

**Issue by bughit** [electron/electron#17470]
* **Electron Version:**
  * 4.1.0
* **Operating System:**
  * win7 sp1 x64

### Expected Behavior
presumably node globals on an isolated context window are not exposed to content, so there's no need to remove them

### Actual Behavior
they are removed

##### index.js
```js
'use strict';

const {app, BrowserWindow} = require('electron'),
      path = require('path');

let win;

app.on('ready', async () => {

  win = new BrowserWindow({
    width: 1440,
    height: 900,
    useContentSize: true,
    webPreferences: {
      nodeIntegration: false,
      contextIsolation: true,
      preload: path.join(__dirname, 'preload.js'),
    },
  });

  win.loadURL('about:blank');
  win.webContents.openDevTools({mode: 'right'});

});
```

##### preload.js
```js
'use strict';

console.log('(global === window) is', global === window);

function logSetImmediate(when) {
  console.log(when, "('setImmediate' in window) is", 'setImmediate' in window);
}

logSetImmediate('during preload');
process.nextTick(() => logSetImmediate('in nextTick callback'));
Promise.resolve().then(() => logSetImmediate('in promise callback'));
setImmediate(() => logSetImmediate('in setImmediate callback'));
setTimeout(() => logSetImmediate('in setTimeout'));
```

##### output
```
(global === window) is true
during preload ('setImmediate' in window) is true
in promise callback ('setImmediate' in window) is true
in nextTick callback ('setImmediate' in window) is true
in setImmediate callback ('setImmediate' in window) is false
in setTimeout ('setImmediate' in window) is false
```

**Comment by bughit** [comment]
@zcbenz Is stripping node globals  from an isolated context needed for security? If not they should be left alone, right?

**Comment by zcbenz** [comment]
Good question, I can't think of a reason not to reserve node globals in isolated context, it seems a good thing to have.

**Comment by miniak** [comment]
Fixed by #17641
